iml-gui/configs/tailwind.config.js

```diff
@@ -211,8 +211,16 @@ module.exports = {
     }
   },
   variants: {
-    display: ["group-hover", "responsive"],
+    display: ["group-hover", "group-focus", "responsive"],
     textColor: ["hover", "group-hover", "focus"]
   },
-  plugins: []
+  plugins: [
+    function({ addVariant }) {
+      addVariant('group-focus', ({ container, separator }) => {
+        container.walkRules(rule => {
+          rule.selector = `.group:focus-within .group-focus\\:${rule.selector.slice(1)}`
```

Comment:
There is some noise:
```
⚠ ｢Critters｣: 45 rules skipped due to selector errors:
  .group:focus-within .group-focus\ -> '.group:focus-within .group-focus\' is not a valid selector
  .group:focus-within .group-focus\ -> '.group:focus-within .group-focus\' is not a valid selector
  .group:focus-within .group-focus\ -> '.group:focus-within .group-focus\' is not a valid selector
  .group:focus-within .group-focus\ -> '.group:focus-within .group-focus\' is not a valid selector
  .group:focus-within .group-focus\ -> '.group:focus-within .group-focus\' is not a valid selector
  .group:focus-within .group-focus\ -> '.group:focus-within .group-focus\' is not a valid selector
  .group:focus-within .group-focus\ -> '.group:focus-within .group-focus\' is not a valid selector
...
```

Reply:
If you update plugins to look like this, it will remove the warnings:

```
  plugins: [
    ({ addVariant, config }) => {
      const prefixClass = function(className) {
        const prefix = config("prefix");
        const getPrefix = typeof prefix === "function" ? prefix : () => prefix;
        return `${getPrefix(`.${className}`)}${className}`;
      };

      const groupPseudoClassVariant = function(pseudoClass) {
        return ({ modifySelectors, separator }) => {
          return modifySelectors(({ selector }) => {
            return selectorParser(selectors => {
              selectors.walkClasses(classNode => {
                classNode.value = `group-${pseudoClass}${separator}${classNode.value}`;
                classNode.parent.insertBefore(
                  classNode,
                  selectorParser().astSync(
                    `.${prefixClass("group")}:${pseudoClass} `
                  )
                );
              });
            }).processSync(selector);
          });
        };
      };

      addVariant("group-focus", groupPseudoClassVariant("focus"));
    }
  ]
```

You will need to add `postcss-selector-parser` as a devDependency and require it at the top of this file as well:

`const selectorParser = require("postcss-selector-parser");`

Reply:
No, the warning is triggered by `focus-within`. It looks like you used `focus` and it worked well. If I change my code to use `focus` the warning will be gone too.

Tailwind uses `focus-within` in other places without [visible] issues. Something else is broken, probably Critters => https://github.com/GoogleChromeLabs/critters/issues/47

Reply:
Hrrm, why do we need focus-within?

Reply:
Because we need show one element, when the other have focus:

```html
<div class="parent">
<span class="input">element to get focus</span>
<span class="popup">popup</span>
</div>
```

I can think of two reliable ways of doing that:

```css
.parent:focus-within .popup {
//...
}
```

or:
```css
.input:focus ~ .popup {
//...
}
```

The latter AFAIK is not possible with  Tailwind. And `:focus` does not propagates to parents.

It is possible to use `.input:focus .popup`, but we would need to make a popup a child of the input, which is odd in general: elements that can be "focus" cannot have children.

Reply:
hrrm, What's strange is I tested this with just focus and it seemed to work fine. I'll try it again.

Reply:
Yeah, it works fine with just focus on the parent. 

![focus-parent](https://user-images.githubusercontent.com/458717/73754143-ec33aa80-4731-11ea-928e-5cfd61218c86.gif)


`:focus-within` is for dealing with focused child selectors anyway (i.e. applying styles to a form when an input is focused), which is the opposite of what we want.

Reply:
Yes, but this feels like a hack, because not all elements, that can be focused, can have children, `input` can't, for instance. Although the only issue we have is that some tool is not updated yet for CSS4 :) `:focus-within` was introduced to overcome limitations of `:focus`, so that the "focus" state of a child were known to parents. And of course it can be used to style sibling, e. g. to highlight an input's label.

Reply:
> Yes, but this feels like a hack, because not all elements, that can be focused, can have children, input can't, for instance.

Sure, but I don't think we're going to be adding a popover to a input element that only opens on focus anytime soon anyway (and if we do, we can tackle that problem when it occurs).

This gets us over the hump of the functionality working the same for popover (which is the purpose of this patch) while not spamming the console with tons of warnings everytime we save a file.

Reply:
There is a logLevel option for Critters and we can suppress warnings, this will not affect functionality. I don't want to taint one thing just because another is lame :) I could question that Critters is useful at all in our case, but it is fine as is.

Reply:
So that chain looks this way: Critters -> [JSDOM](https://github.com/jsdom/jsdom/issues/2026#issuecomment-339050204) -> [CSSOM](https://github.com/NV/CSSOM/).

Reply:
It's sort of trading one workaround for another, but I can live with it. Should probably put the issue next to the logLevel bump so it can be tracked and revered in the future

Reply:
I think I will open an issue for Critters.